src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/readme.md

```diff
@@ -42,14 +44,14 @@ The following example shows a sample JSON response:
-| id      | Mandatory | string | A unique identifier for the specific entitlement issued to the application. <br/> Multiple entitlement requests for the same application from the same compute node may (but are not required to) return the same identifier. <br/> Entitlement requests from different compute nodes will not return duplicate identifiers. <br/> Clients should make no assumptions about the structure of the `id` as it may change from release to release. |
-| vmid    | Mandatory | string | The unique [virtual machine identifier](https://azure.microsoft.com/blog/accessing-and-using-azure-vm-unique-id/) of the entitled Azure virtual machine. <br/> Clients may optionally check this matches the actual virtual machine identifier for the host machine.                                                                                                                                                                          |
+| Element | Required  |   Type   |                                                                                                                                                                                                                   Description                                                                                                                                                                                                                   |
+| ------- | --------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| id      | Mandatory | string   | A unique identifier for the specific entitlement issued to the application. <br/> Multiple entitlement requests for the same application from the same compute node may (but are not required to) return the same identifier. <br/> Entitlement requests from different compute nodes will not return duplicate identifiers. <br/> Clients should make no assumptions about the structure of the `id` as it may change from release to release. |
+| expiry  | Mandatory | DateTime | The expiry timestamp of the token; further verification requests after this instant will be declined.                                                                                                                                                                                                                                                                                                                                           |
```

Comment:
If the token has expired should the error response indicate as much?  E.g. 400 with a suitable error body?  Otherwise it's kind of "well what went wrong, the token looks okay to me".  I'm sure someone will argue that this leaks security information to someone trying to misuse the token, but it also feels like it provides important context if something goes wrong in the legitimate production scenario.

Reply:
In the production scenario, we get detailed logs from TLSFE that tell us exactly what happened; in the dev/test scenario, the `sestest` server gives detailed diagnostics on the console.

Also, a declined entitlement is a 403 - we only return 400 if something goes wrong.